test/helpers/data.go

```diff
@@ -0,0 +1,39 @@
+package helpers
+
+import (
+	"math/rand"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	letterBytes   = "abcdefghijklmnopqrstuvwxyz"
+	randSuffixLen = 8
+	sep = "-"
+)
+
+var (
+	r        *rand.Rand
+	rndMutex *sync.Mutex
+	once     sync.Once
+)
+
+func initSeed() {
+	seed := time.Now().UTC().UnixNano()
+	r = rand.New(rand.NewSource(seed))
+	rndMutex = &sync.Mutex{}
+}
+
+func AppendRandomString(prefix string) string {
+	once.Do(initSeed)
+	suffix := make([]byte, randSuffixLen)
+
+	rndMutex.Lock()
+	for i := range suffix {
+		suffix[i] = letterBytes[r.Intn(len(letterBytes))]
+	}
+	rndMutex.Unlock()
```

Comment:
Please `defer` this right after getting the Lock above. The minimal gain in concurrency is not worth the risk of a panic in between I think.

Reply:
Done